src/amo/components/ErrorPage/NotFound/index.js

```diff
@@ -1,30 +1,47 @@
+/* @flow */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { compose } from 'redux';
 import NestedStatus from 'react-nested-status';
 
 import SuggestedPages from 'amo/components/SuggestedPages';
+import {
+  ERROR_ADDON_DISABLED_BY_ADMIN, ERROR_ADDON_DISABLED_BY_DEV,
+} from 'core/constants';
 import translate from 'core/i18n/translate';
 import { sanitizeHTML } from 'core/utils';
 import Card from 'ui/components/Card';
+import type { I18nType } from 'core/types/i18n';
 
 import 'amo/components/ErrorPage/ErrorPage.scss';
 
+type Props = {|
+  errorCode?: string,
+  i18n: I18nType,
+|};
 
-export class NotFoundBase extends React.Component {
-  static propTypes = {
-    i18n: PropTypes.object.isRequired,
-  }
-
+export class NotFoundBase extends React.Component<Props> {
   render() {
-    const { i18n } = this.props;
+    const { errorCode, i18n } = this.props;
 
     const fileAnIssueText = i18n.sprintf(i18n.gettext(`
       If you followed a link from somewhere, please
       <a href="%(url)s">file an issue</a>. Tell us where you came from and
       what you were looking for, and we'll do our best to fix it.`),
     { url: 'https://github.com/mozilla/addons-frontend/issues/new/' });
 
+    let explanation;
+    if (errorCode === ERROR_ADDON_DISABLED_BY_DEV) {
+      explanation =
+        i18n.gettext('This add-on has been removed by its author.');
+    } else if (errorCode === ERROR_ADDON_DISABLED_BY_ADMIN) {
+      explanation =
+        i18n.gettext('This add-on has been disabled by an administrator.');
```

Comment:
Should we just say "disabled by Mozilla"? I think that makes more sense and is a bit clearer.

Reply:
I got the copy from the old site. I was thinking about it and I like how administrator is a bit more AMO-focused and inclusive, meaning we have volunteers who may be helping to disable things (well, I know we have volunteer reviewers but I'm not sure about admins).